File: HelloApp/app/src/main/java/com/example/helloapp/MainActivity.java
package com.example.helloapp;

import android.content.Context;
import android.os.Bundle;
import android.view.ContextMenu;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class MainActivity extends AppCompatActivity implements View.OnClickListener{

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        View.OnClickListener onClickListener = new View.OnClickListener() {

            @Override
            public void onClick(View view) {
                Toast.makeText(MainActivity.this, "반가워", Toast.LENGTH_SHORT).show();

            }
        } ;

        TextView txt1 = findViewById(R.id.txt1);
        registerForContextMenu(txt1);

        FloatingActionButton fab = findViewById(R.id.fab);
        // fab.setOnClickListener(onClickListener);
        // registerForContextMenu(fab);
        // fab.setOnCreateContextMenuListener(onCreateContextMenu());
 /*       fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show();
            }
        });*/
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_option, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onCreateContextMenu(ContextMenu menu,
                                    View v,
                                    ContextMenu.ContextMenuInfo menuInfo)
    {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu_option, menu);
    }


    @Override
    public void onClick(View v) {

    }







    protected class MyOnClick extends View
    {

        public MyOnClick(Context context) {
            super(context);
        }

        @Override
        public void setOnClickListener(@Nullable OnClickListener l) {
            super.setOnClickListener(l);
        }
    }
}

File: Naver_Kt_01/app/src/main/java/com/callor/naver/MainActivity.kt
package com.callor.naver

import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import android.view.inputmethod.EditorInfo
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.StaggeredGridLayoutManager
import com.callor.naver.adapter.MovieAdapter
import com.callor.naver.config.NaverSecur
import com.callor.naver.domain.Movie
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.content_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        val txtSearch = txt_search
        val recyclerView = movie_list

        // txtSearch.setOnEditorActionListener { v, actionId, event ->
        // 사용하지 않은 파라메터 이름 _ 로 변경
        txtSearch.setOnEditorActionListener { _, actionId, _ ->
            if(actionId == EditorInfo.IME_ACTION_SEARCH){

                val strSearch = txtSearch.text.toString()
                val naverCall = RetrofitClient.getApiService().getSearch(NaverSecur.NAVER_ID,NaverSecur.NAVER_SEC,"movie.json",strSearch)
                naverCall.enqueue(object : Callback<Movie>{

                    override fun onResponse(call: Call<Movie>, response: Response<Movie>) {
                        val mList: Array<Movie.Items>? = response.body()?.items

                        val movieAdapter = mList?.let { MovieAdapter(it) }
                        recyclerView.adapter = movieAdapter

                        val layoutManager =
                            StaggeredGridLayoutManager(1, StaggeredGridLayoutManager.VERTICAL)
                        recyclerView.layoutManager = layoutManager

                    }

                    override fun onFailure(call: Call<Movie>, t: Throwable) {
                        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
                    }

                })
                true
            } else {
                false
            }
        }


    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }
}

